src/distilabel/pipeline.py

```diff
@@ -623,20 +625,39 @@ def _generate(  # noqa: C901
                         progress_callback_func=generation_progress_func,
                     )
                     generations.extend(batch_generations)
+                    # If we have both generator and labeller, save only after the labeller
+                    if checkpoint_strategy and (self.labeller is None):
+                        if checkpoint_strategy.do_checkpoint(batch_i * batch_size):
+                            logger.info(f"Saving dataset up to batch {batch_i}...")
+                            ds = self._build_dataset(
+                                dataset,
+                                generations=generations,
+                                labels=labels,
+                                batch_size=batch_size,
+                            )
+                            ds.save_to_disk(
+                                checkpoint_strategy.path,
+                                **checkpoint_strategy.extra_kwargs,
+                            )
```

Comment:
I think this block within the `if self.generator is not None` and condition and the one below within the `if self.labeller is not None` condition could be the same, right? We could just simply:

```python
for batch_i, rows in enumerate(dataset.iter(batch_size=batch_size), start=1):
    if self.generator is not None: ...
    if self.labeller is not None: ...
    if checkpoint_strategy and checkpoint_strategy.do_checkpoint(batch_i * batch_size): # do checkpoint
```

Reply:
yes, thanks!